files/bootstrap.sh

```diff
@@ -55,6 +56,11 @@ while [[ $# -gt 0 ]]; do
             shift
             shift
             ;;
+        --kubelet-extra-json-config)
```

Comment:
* You've defined the arg as `--kubelet-extra-json-args` in the helper on #L23. Can we instead have this as `--kubelet-extra-json-config-file` and take a pointer to a file instead of the file contents?
* You probably also want to default `KUBELET_EXTRA_JSON_CONFIG` like on #L126, otherwise it might fail with this variable not being bound in the `if -n` check later on.

Reply:
Hm, I'm not sure we can. I remember something about this, but we pass it in as definitive data. I don't think we can make it be on the instance when this is being executed.

Reply:
My only nit is I'd call this `--kubelet-extra-config`/`KUBELET_EXTRA_CONFIG`. We don't need to mention the format of the config file here, and this should have the same semantics as `--kubelet-extra-args`.

Reply:
Create the file with what content? :)

Reply:
> eksctl is no longer constructing userdata

Maybe I'm misunderstanding, but isn't eksctl creating userData with `/etc/eks/bootstrap.sh $args`? I don't think I quite understand why eksctl cannot take in the json input for `kubeletExtraConfig`, put it in a file and then provide it to the bootstrap script. 

If instead, it an eksctl customer that's fully authoring their UserData (including their bootstrap script), then I'd expect them to do - 
```
cat <<EOF > /etc/eks/kubelet-custom-config.json
{
  "kubeReserved": 240m
}
EOF
/etc/eks/bootstrap.sh --kubelet-extra-config /etc/eks/kubelet-custom-config.json $otherArgs
```

I understand that the eksctl API accepts additional kubelet config as JSON / YAML input if I'm [reading this right](https://github.com/weaveworks/eksctl/issues/4459), but I don't think we need to force the bootstrap script's interface to also match that semantic.

Reply:
It doesn't work, because anything we do just gets overwritten ultimately in this script. Whatever we do outside, the value just gets clobbered over with some defaults.

The reason we don't do the first part is that using `kubelet-extra-config` is cumbersome and overly complex for simple values and passing in some json data. :)